ulib/FStar.Buffer.fst

```diff
@@ -822,16 +822,16 @@ val createL: #a:Type0 -> init:list a -> StackInline (buffer a)
      /\ frameOf b == h0.tip
      /\ Map.domain h1.h == Map.domain h0.h
      /\ modifies_0 h0 h1
-     /\ as_seq h1 b == Seq.of_list init
+     /\ as_seq h1 b == Seq.seq_of_list init
      /\ q #a len b))
 #set-options "--initial_fuel 1 --max_fuel 1" //the normalize_term (length init) in the pre-condition will be unfolded
 	                                     //whereas the L.length init below will not
 let createL #a init =
   let len = UInt32.uint_to_t (FStar.List.Tot.length init) in
   let s = Seq.of_list init in
-  lemma_of_list_length s init;
+  lemma_seq_list_bij s;
   let content: reference (s:seq a{Seq.length s == v len}) =
-    salloc (Seq.of_list init) in
```

Comment:
There is an impact on performance in changing `of_list` by `seq_of_list`. 
`of_list` runs in O(1), while `seq_of_list l` runs in O(length l).  I'm not sure this matters though.
Regardless, `Seq.Base.of_list` and `Seq.Properties.seq_of_list` are redundant. We should either choose to maximize abstraction and keep only the derived `Seq.Properties.seq_of_list`, or maximize performance and keep only `Seq.Base.of_list`.

Reply:
`Seq.Properties.seq_of_list` is more convenient to use, because there is a complementary `Seq.Properties.seq_to_list` and a bijectivity lemma.